### PR TITLE
Convert metadata dom 20190320

### PR DESCRIPTION
**Pull request by climbfuji**
Code changes to make the converter work with GFS_typedefs.F90, in particular handle !defined(X) preprocessor statements. Also: changes to top-level convert_metadata.py script that defines the logger, partially overlapping with what @gold2718 just pushed a few minutes ago.

### CODE REVIEW

File: scripts/convert_metadata.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 #!/usr/bin/env python
+
 # Python library imports
 import sys
 import os.path
 import re
 import logging
 from collections import OrderedDict
+import logging
```
Comment (gold2718) [inline scripts/convert_metadata.py:8]: Why is this line necessary?

File: scripts/convert_metadata.py [hunk 2]
```diff
@@ -367,7 +367,7 @@ def convert_file(filename_in, filename_out, metadata_filename_out, logger):
 def usage(cmd):
     print("Usage:")
     print("{} <source_file> <target_file>".format(cmd))
-    print("{} <source_file> [ <source_file> ...] [ <target_directory>".format(cmd))
+    print("{} <source_file> [ <source_file> ...] [ <target_directory> ]".format(cmd))
```
Comment (gold2718) [inline scripts/convert_metadata.py:369]: oops, thanks

**Review by gold2718 — CHANGES_REQUESTED** [review]
Do we need two import logging lines? If not, please remove one.

File: scripts/convert_metadata.py (hunk 1 above)
Comment (climbfuji) [reply scripts/convert_metadata.py:8]: Uups, I had introduced the logger earlier on and then merged in your recent commit that did the same.

**Review by gold2718 — APPROVED** [review]

**Comment by climbfuji** [comment]
I think you need to merge this, @gold2718

**Comment by gold2718** [comment]
Code changes to make the converter work with GFS_typedefs.F90, in particular handle !defined(X) preprocessor statements. Also: changes to top-level convert_metadata.py script that defines the logger, partially overlapping with what @gold2718 just pushed a few minutes ago.
